src/ddmd/cppmangle.d

```diff
@@ -815,10 +815,12 @@ public:
             case Tuns32:                c = 'j';        break;
             case Tfloat32:              c = 'f';        break;
             case Tint64:
-                c = (Target.c_longsize == 8 ? 'l' : 'x');
+                // Mangle like target's stdint.h int64_t
+                c = (Target.c_longsize == 4 || global.params.isOSX ? 'x' : 'l');
```

Comment:
This should be turned into a target hook. The frontend should not be asking questions such as "Am I targeting osx?". I think we already have a mangle type hook already present, the logic can be put there.

Reply:
https://github.com/dlang/dmd/blob/5494dba193d16d6e126af8a12d37567b4d245086/src/ddmd/target.d#L459

Comment would need to be updated to reflect that if the target requires mangling in a specific way, then handle it here.